src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5474,8 +5474,9 @@ std::optional<ProjectionCandidate> MergeTreeData::getQueryProcessingStageWithAgg
         // If optimize_aggregation_in_order = true, we need additional information to transform the projection's pipeline.
         auto attach_aggregation_in_order_info = [&]()
         {
-            for (const auto & key : keys)
+            for (const auto & desc : select.getQueryAnalyzer()->aggregationKeys())
```

Comment:
What's the purpose of this change?

Reply:
Order should be preserved for in order aggregation (this has been covered in tests).

See https://github.com/ClickHouse/ClickHouse/pull/37469/commits/a966cfc682b41ec6fe8153bd35f61ab9d4cf7567.